syntax/abstract.mjs

```diff
@@ -1,36 +1,52 @@
 import * as FTL from "./ast.mjs";
+import {always, never} from "../lib/combinators.mjs";
 
 export function list_into(Type) {
     switch (Type) {
         case FTL.CallExpression:
-            return ([callee, args = []]) =>
-                new Type(callee, args);
+            return ([callee, args]) => {
+                let posargs = [];
+                let kwargs = [];
+                for (let arg of args) {
+                    if (arg instanceof FTL.NamedArgument) {
+                        kwargs.push(arg);
```

Comment:
In the future, we could add a semantic check here to forbid duplicate named arguments. I filed #114 to track this.